packages/react-components/src/shared/src/mergeProps.ts

```diff
@@ -107,12 +113,16 @@ function merge(props, newProps) {
     return props;
 }
 
-export function mergeProps(...args) {
+// taken from: https://stackoverflow.com/questions/51603250/typescript-3-parameter-list-intersection-type/51604379#51604379
+type TupleTypes<T> = { [P in keyof T]: T[P] } extends { [key: number]: infer V } ? V : never;
+type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends ((k: infer I) => void) ? I : never;
+
+export function mergeProps<T extends Record<string, any>[]>(...args: T) {
```

Comment:
Does the return type are implicits?

Reply:
Ya, since we cast it the last line as a type, typescript infers the return value of this method